spectree/plugins/falcon_plugin.py

```diff
@@ -192,3 +214,64 @@ def bypass(self, func, method):
         if isinstance(func, partial):
             return True
         return inspect.isfunction(func)
+
+
+class FalconAsgiPlugin(FalconPlugin):
+    """Light wrapper around default Falcon plug-in to support Falcon 3.0 ASGI apps"""
+
+    ASYNC = True
+    OPEN_API_ROUTE_CLASS = OpenAPIAsgi
+    DOC_PAGE_ROUTE_CLASS = DocPageAsgi
+
+    async def request_validation(self, req, query, json, headers, cookies):
+        if query:
+            req.context.query = query(**req.params)
+        if headers:
+            req.context.headers = headers(**req.headers)
+        if cookies:
+            req.context.cookies = cookies(**req.cookies)
```

Comment:
To keep it consistent with other parts, it should be better to use `.parse_obj()`.

Reply:
Good call! I caught that on one of the `json` calls, but missed it on these. I'll get that updated. I'm also having some trouble getting the async stuff working on a local testing project, so I'll let you know when I've got that sorted out.